### PR TITLE
[Enhancement][RHEL/7] Add missing permission checks

### PR DESCRIPTION
**Pull request by redhatrises**
- Add missing dir perms checks
- Adding missing file perms checks
- Add root no-dot-in-path check
- Add missing unowned and ungrouped checks

### CODE REVIEW

File: shared/oval/dir_perms_etc_httpd_conf.xml [hunk 1]
```diff
@@ -3,10 +3,10 @@
...
       <affected family="unix">
-        <platform>Red Hat Enterprise Linux 6</platform>
+        <platform>multi_platform_rhel</platform>
       </affected>
       <description>Directory permissions for /etc/httpd/conf/ should be set to 0750 (or stronger).</description>
-      <reference source="JL" ref_id="20140403" ref_url="test_attestation"/>
+      <reference source="galford" ref_id="20160120" ref_url="test_attestation"/>
     </metadata>
     <criteria>
       <criterion test_ref="test_dir_perms_etc_httpd_conf" />
```
Comment (iankko) [inline shared/oval/dir_perms_etc_httpd_conf.xml]: The full form of the criteria is as below:
```
    <criteria>
      <criterion test_ref="test_dir_perms_etc_httpd_conf" />
    </criteria>
```
There should be added check if the ```httpd``` package is actually installed (because it returns ```fail``` even in the case ```httpd``` isn't installed on the system). This is NOT to say there's bug in this OVAL, but that it should be enhanced for both RHEL-6 && RHEL-7.

File: shared/oval/dir_perms_var_log_httpd.xml [hunk 2]
```diff
@@ -3,10 +3,10 @@
...
       <affected family="unix">
-        <platform>Red Hat Enterprise Linux 6</platform>
+        <platform>multi_platform_rhel</platform>
       </affected>
       <description>Directory permissions for /var/log/httpd should be set to 0700 (or stronger).</description>
-      <reference source="JL" ref_id="20140403" ref_url="test_attestation"/>
+      <reference source="galford" ref_id="20160120" ref_url="test_attestation"/>
     </metadata>
     <criteria>
       <criterion test_ref="test_dir_perms_var_log_httpd" />
```
Comment (iankko) [inline shared/oval/dir_perms_var_log_httpd.xml]: Ditto, the check for 'package_httpd_installed' should be added there.

File: shared/oval/dir_perms_etc_httpd_conf.xml (hunk 1 above)
Comment (iankko) [inline shared/oval/dir_perms_etc_httpd_conf.xml]: When httpd is installed, verified the oval check output is as expected ('fail' for 755, 'pass' for 750).

File: shared/oval/dir_perms_var_log_httpd.xml (hunk 2 above)
Comment (iankko) [inline shared/oval/dir_perms_var_log_httpd.xml]: When httpd is installed, OVAL works fine.

File: shared/oval/file_permissions_httpd_server_conf_files.xml [hunk 3]
```diff
@@ -4,10 +4,10 @@
...
       <title>Verify Permissions On Apache Web Server Configuration Files</title>
       <affected family="unix">
-        <platform>Red Hat Enterprise Linux 6</platform>
+        <platform>multi_platform_rhel</platform>
       </affected>
       <description>The /etc/httpd/conf/* files should have the appropriate permissions (0640 or stronger).</description>
-      <reference source="JL" ref_id="20140403" ref_url="test_attestation"/>
+      <reference source="galford" ref_id="20160120" ref_url="test_attestation"/>
     </metadata>
     <criteria>
```
Comment (iankko) [inline shared/oval/file_permissions_httpd_server_conf_files.xml]: Again, this should depend on 'package_httpd_installed' OVAL.

File: shared/oval/file_permissions_httpd_server_conf_files.xml [hunk 4]
```diff
@@ -4,10 +4,10 @@
     <metadata>
       <title>Verify Permissions On Apache Web Server Configuration Files</title>
       <affected family="unix">
-        <platform>Red Hat Enterprise Linux 6</platform>
+        <platform>multi_platform_rhel</platform>
       </affected>
       <description>The /etc/httpd/conf/* files should have the appropriate permissions (0640 or stronger).</description>
```
Comment (iankko) [inline shared/oval/file_permissions_httpd_server_conf_files.xml]: Besides that from testing this OVAL fails, when the permissions of ```/etc/httpd/conf/*``` files are 600 (IOW stronger than 0640). This is because the state below requires exactly ```0640```. It should be modified to allow stronger perms too.

File: shared/oval/dir_perms_world_writable_sticky_bits.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
     <metadata>
       <title>Verify that All World-Writable Directories Have Sticky Bits Set</title>
```
Comment (iankko) [inline shared/oval/dir_perms_world_writable_sticky_bits.xml]: This works fine (verified).

File: shared/oval/dir_perms_world_writable_system_owned.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<def-group>
+  <definition class="compliance" id="dir_perms_world_writable_system_owned" version="1">
```
Comment (iankko) [inline shared/oval/dir_perms_world_writable_system_owned.xml]: Ditto (this seem to be working fine).

File: shared/oval/file_permissions_unauthorized_world_writable.xml [hunk 7]
```diff
@@ -3,7 +3,7 @@
     <metadata>
       <title>Find Unauthorized World-Writable Files</title>
```
Comment (iankko) [inline shared/oval/file_permissions_unauthorized_world_writable.xml]: This works fine (verified).

File: shared/oval/file_permissions_ungroupowned.xml [hunk 8]
```diff
@@ -3,7 +3,7 @@
     <metadata>
       <title>Find files unowned by a group</title>
```
Comment (iankko) [inline shared/oval/file_permissions_ungroupowned.xml]: This works fine (verified).

File: shared/oval/no_files_unowned_by_user.xml [hunk 9]
```diff
@@ -3,7 +3,7 @@
     <metadata>
       <title>Find files unowned by a user</title>
```
Comment (iankko) [inline shared/oval/no_files_unowned_by_user.xml]: This works fine (verified).

File: shared/oval/root_path_no_dot.xml [hunk 10]
```diff
@@ -3,7 +3,7 @@
     <metadata>
       <title>Ensure that No Dangerous Directories Exist in Root's Path</title>
```
Comment (iankko) [inline shared/oval/root_path_no_dot.xml]: This one seems to be passing anytime (even if ```~/.bashrc``` contains something like:
```
PATH=$PATH:
```
But this is problem of RHEL-6 implementation => better to be handled within separate PR).

**Comment by redhatrises** [comment]
@iankko 
> Ditto, the check for 'package_httpd_installed' should be added there.

Done.

> Besides that from testing this OVAL fails, when the permissions of /etc/httpd/conf/* files are 600 (IOW stronger than 0640). This is because the state below requires exactly 0640. It should be modified to allow stronger perms too.

Done.

**Comment by iankko** [comment]
Thanks. Will try shortly. For the last ```root_path_no_dot``` rule -- don't spend much time on that one. This can be done in future PRs -- I will just file a ticket for that so we don't forget about it.

File: shared/oval/dir_perms_etc_httpd_conf.xml [hunk 11]
```diff
@@ -3,12 +3,14 @@
...
-      <reference source="JL" ref_id="20140403" ref_url="test_attestation"/>
+      <reference source="galford" ref_id="20160120" ref_url="test_attestation"/>
     </metadata>
-    <criteria>
+    <criteria operator="OR">
+      <extend_definition comment="httpd not present or in use"
+      definition_ref="package_httpd_removed" />
       <criterion test_ref="test_dir_perms_etc_httpd_conf" />
     </criteria>
   </definition>
```
Comment (iankko) [inline shared/oval/dir_perms_etc_httpd_conf.xml]: Here the mode check should also allow stronger permissions (sorry for not catching that immediately). Just remove all the 'true'  lines in the ```state_dir_perms_etc_httpd_conf``` state below (since they are irrelevant). When they are set the mode is 0750, if they aren't set the mode is stronger. What matters being all those "false" valued lines in the state below.

**Comment by redhatrises** [comment]
> sorry for not catching that immediately

No worries.

> chmod 0600 /etc/httpd/conf/*

Should this also exist for  `/etc/httpd/conf.d` as well?

**Comment by iankko** [comment]
@redhatrises 
> Should this also exist for /etc/httpd/conf.d as well?

I would say so. From the XCCDF <rationale> of that rule:
```
<rationale>
Access to the web server's configuration files may allow an unauthorized user or attacker
to access information about the web server or alter the server's configuration files.
</rationale>
```
IMHO this rule should cover all httpd configuration files (thus also those possibly located in ```/etc/httpd/conf.d```, not just explicitly ```/etc/httpd/conf``` directory). IMO the rule has been created in times of RHEL-5 yet, where there hasn't been ```httpd/conf.d``` location, yet. So IMHO we should enhance to current status.

@shawndwells What are your thoughts on the above?

**Comment by iankko** [comment]
* The ```root_path_no_dot``` issue is reported under -- https://github.com/OpenSCAP/scap-security-guide/issues/980
* The requirement to add two new perms ```0750``` directory checks for ```dir_perms_etc_httpd_conf_d``` and ```dir_perms_etc_httpd_modules_d``` is tracked under -- https://github.com/OpenSCAP/scap-security-guide/issues/985 , finally
* The requirement to enhance the existing XCCDF, OVAL && remediations for ```file_permissions_httpd_server_conf_files``` to check ```0640 (or stronger)``` permissions also on configuration files possibly located in ```etc/httpd/conf.d``` and ```/etc/httpd/conf.modules.d``` paths is now tracked under -- https://github.com/OpenSCAP/scap-security-guide/issues/986 issue.

Retested the latest version, and verified both of the checks:
* ```dir_perms_etc_httpd_conf.xml ``` and
* ```file_permissions_httpd_server_conf_files.xml```

now allow setting of also stronger permissions than the requirement.

ACK from me for merging this.

@redhatrises Thanks, Gabe!
